database/migrations/2020_03_15_122454_add_metadata_fields_to_users.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class AddMetadataFieldsToUsers extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        DB::connection()->getDoctrineSchemaManager()->getDatabasePlatform()->registerDoctrineTypeMapping(
+            'enum',
+            'string'
+        );
+
+        Schema::table('users', static function (Blueprint $table): void {
+            $table->string('create_reason', 255)->default('cas_login');
+            $table->boolean('has_ever_logged_in')->default(true);
```

Comment:
Is this necessary if you have the `last_login` timestamp?

Reply:
Tbh not really. The updated timestamp will always match this since it takes a database write to set that field.

Reply:
You can’t backfill the time stamp in the migration unless there’s an arbitrary date that means unknown but has logged in.

Reply:
If the has logged in flag is set then the last updated timestamp is the last login timestamp, otherwise they haven't logged in ever.